pkg/clusters/addons/kong/kong.go

```diff
@@ -173,9 +214,10 @@ func (a *Addon) Ready(ctx context.Context, cluster clusters.Cluster) (waitForObj
 		return
 	}
 
-	for _, deployment := range deployments.Items {
+	for i := 0; i < len(deployments.Items); i++ {
```

Comment:
can you use `range` instead of a C-style `for` here? won't it be more readable that way?

Reply:
This way is safer, and helps avoid pulling pointers to loop variables, the change was intentional and in response to lint errors (because we did have an actual issue).

I would like to keep this, and in general I don't think we should be afraid of C-style `for` loops, though when we don't need pointers to the loop variables `range` is great.